modules/katamari/src/main/ts/ephox/katamari/api/Adt.ts

```diff
@@ -50,21 +50,15 @@ const generate = function <T = Record<string, (...data: any[]) => Adt>> (cases:
     //
     // constructor for key
     //
-    adt[key] = function (): Adt {
-      const argLength = arguments.length;
+    adt[key] = (...args: any[]): Adt => {
+      const argLength = args.length;
 
       // validation
       if (argLength !== value.length) {
         throw new Error('Wrong number of arguments to case ' + key + '. Expected ' + value.length + ' (' + value + '), got ' + argLength);
       }
 
-      // Don't use array slice(arguments), makes the whole function unoptimisable on Chrome
-      const args = new Array(argLength);
```

Comment:
Note:  Remove this here and in `Merger` as using ES6 rest parameters achieves the same result (ie converting `arguments` to an array).